lib/set.rb

```diff
@@ -812,23 +812,51 @@ def join(separator=nil)
     to_a.join(separator)
   end
 
+  def self.__inspect_supports_identity_set?
```

Comment:
This is checked by the corresponding PR in `ostruct`:

https://github.com/ruby/ostruct/pull/58/files#diff-fcafc50c8e2ed25a9c3851150cc0bdb9c9282c201d80574d126e65e8ea84e946R391